src/compat/event_loop.rs

```diff
@@ -65,10 +65,13 @@ pub fn spawn_event_loop<UID: Uid>(
 
         match try() {
             Ok((mut core, mut service_state)) => {
+                let handle = core.handle();
                 let (tx, rx) = mpsc::unbounded::<ServiceCommand<UID>>();
                 unwrap!(result_tx.send(Ok(tx)));
                 unwrap!(core.run({
-                    rx.for_each(move |cb| Ok(cb.call_box(&mut service_state)))
+                    rx
+                    .for_each(move |cb| Ok(cb.call_box(&mut service_state)))
+                    .and_then(move |()| Timeout::new(Duration::from_millis(200), &handle).infallible())
```

Comment:
Just curious, what does that buy us?

Reply:
It inelegantly fixes the bug which I think was causing `compat_api::bootstrap_using_hard_coded_contacts` to fail. It you close a uTP connection then immediately shutdown the event loop it's possible for the FIN packet to not get sent. Keeping the event loop alive a little while longer gives some time for the socket to flush itself.